### PR TITLE
cleaned repo and created adducts.R

### PR DESCRIPTION
**Pull request by michaelwitting**
@jorainer I cleaned up all the stuff and just added file `adducts.R` containing all functions related to adducts

### CODE REVIEW

File: R/adducts.R [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
+  # retrieve multiplicative and additive part for calculation
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  exact_mass <- (ion_mass - mass_add) / mass_multi
+  
+  return(exact_mass)
+  
+}
+
```
Comment (jorainer) [inline R/adducts.R]: I would suggest to rename the function `adduct_rules_all` into `adductRules` and add a parameter `polarity = c("positive", "negative")` to it. Then we can export only this function and keep the other two (`adduct_rules_neg` and `adduct_rules_pos`) as internal functions.

File: R/adducts.R [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
-#' 
-#' 
-#' @author Michael Witting, \email{michael.witting@@helmholtz-muenchen.de}
-#'
-#' @seealso \code{\link{adduct_rules_pos}}
-#' @seealso \code{\link{adduct_rules_neg}}
-#' @seealso \code{\link{get_adduct_names}}
-#'
-#' @export
+mass2mz <- function(exact_mass, adduct) {
```
Comment (jorainer) [inline R/adducts.R]: please change parameter `exact_mass` to `x`. Also, it is usually better to provide a defaults for parameters if possible. For `adduct` you could use `adduct = names(adductRules())`.

File: R/adducts.R [hunk 3]
```diff
@@ -1,18 +1,47 @@
...
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  ion_mass <- exact_mass * mass_multi + mass_add
+  
+  return(ion_mass)
+  
+}
+
+mz2mass <- function(ion_mass, adduct) {
```
Comment (jorainer) [inline R/adducts.R]: also here, please change `ion_mass` to `x` and use a default for the `adduct`.

File: R/adducts.R [hunk 4]
```diff
@@ -1,18 +1,47 @@
...
+    
+  }
+  
+  # retrieve multiplicative and additive part for calculation
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  ion_mass <- exact_mass * mass_multi + mass_add
+  
+  return(ion_mass)
```
Comment (jorainer) [inline R/adducts.R]: `return` should not be required. The default in `R` is to simply use here `ion_mass`.

File: R/adducts.R [hunk 5]
```diff
@@ -1,18 +1,47 @@
...
+    
+  }
+  
+  # retrieve multiplicative and additive part for calculation
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  exact_mass <- (ion_mass - mass_add) / mass_multi
+  
+  return(exact_mass)
```
Comment (jorainer) [inline R/adducts.R]: also here, please use just `exact_mass` and remove the `return`.

**Comment by jorainer** [comment]
BTW, you can obviously work directly in the rformassspectrometry/MetaboCoreUtiles:devel_michael branch!

**Comment by michaelwitting** [comment]
Ah... removed the wrong repo. Can you perform this merge and I will do the changes afterwards? The forking messed up a few things in my home office...

**Comment by jorainer** [comment]
Sure.

**Comment by michaelwitting** [comment]
Thx... Will make the changes soonish!

**Pull request merged**